src/aggregate/aggregate_exec.c

```diff
@@ -161,8 +161,18 @@ static int sendChunk(AREQ *req, RedisModuleCtx *outctx, size_t limit) {
   RedisModule_ReplyWithArray(outctx, REDISMODULE_POSTPONED_ARRAY_LEN);
 
   rc = rp->Next(rp, &r);
-  RedisModule_ReplyWithLongLong(outctx, req->qiter.totalResults);
+  if (rc == RS_RESULT_TIMEDOUT) {
+    if (!(req->reqflags & QEXEC_F_IS_CURSOR)) {
+      RedisModule_ReplyWithSimpleString(outctx, "Timeout limit was reached");
+    } else {
+      rc = RS_RESULT_OK;
```

Comment:
With cursor it's complicated, we do not want to get a timeout for idle cursors, maybe we can reset the start time on cursor read?